LRU0.java

```diff
@@ -0,0 +1,64 @@
+public class Solution {
+
+    // @param capacity, an integer
+    class Node{
+        int key, value;
+        Node prev, next;
+        Node(int k, int v){
+            key = k;
+            value = v;
+            next = null;
+            prev = null;
+        }
+    }
+    HashMap<Integer,Node> map;
+    private int capacity;
+    private Node tail, head;
+    public Solution(int capacity) {
+        this.capacity = capacity;
+        map = new HashMap<>();
+        head = new Node(-1, -1);
+        tail = new Node(-1, -1);
+        head.next = tail;
+        tail.prev = head;
+    }
+    public void movetohead(int key){
+        if (!map.containsKey(key)) return;
+        Node node = map.get(key);
+        if(node.prev != null && node.next != null){
+            node.prev.next = node.next;
+            node.next.prev = node.prev;
+        }
+        node.next = head.next;
+        head.next.prev = node;
+        node.prev = head;
+        head.next = node;
+    }
+    // @return an integer
+    public int get(int key) {
```

Comment:
Use private since this method can only be accessed by LRUCache.class. Use private methods unless you need to override a public API